qcodes/instrument/base.py

```diff
@@ -672,20 +677,21 @@ def find_instrument(cls, name: str,
             TypeError: If a specific class was requested but a different
                 type was found.
         """
+        internal_instrument_class = instrument_class or Instrument
+
         if name not in cls._all_instruments:
             raise KeyError(f"Instrument with name {name} does not exist")
-        ins = cls._all_instruments[name]()
+        ins = cast(T, cls._all_instruments[name]())
```

Comment:
Not sure if there is a way to get rid of this cast

Reply:
I think its fair that it is here since this method anyway checks for the type of the class

Reply:
but perhaps better at the end of the method ?

Reply:
yes, better at the end, because actually on this line it can still be None because the instrument might have been closed but the  record of its instance is (for some reason) still in the `_all_instruments`.